### DISCUSSION TITLE
Is it possible to trigger detect event via API?

### DISCUSSION THREAD

**Discussion by luisiam** [blakeblackshear/frigate#10158] [Ideas]
Version 0.13 adds the ability to create event manually via the API which is great because I can rely on the camera's motion detection to create a frigate event and save some resources from my server. However, the current API only supports the creation of the entire event. Is it possible to trigger object detection via the API and the frigate will run object detection and create the event? That way I can skip the motion detection part in the server while keeping the excellent object detection in frigate.

Now:
Camera Motion Detected --> Frigate Event Create API --> Event Created

Proposal:
Camera Motion Detected --> Frigate Trigger Object Detection API (New) --> Frigate Objection Detection --> Event Created

**Comment by NickM-27** [comment]
Frigate object detection does not scan the whole image and I don't see a reason for this over the existing implementation. If motion occurs then frigate will look for objects,

**Comment by luisiam** [comment]
My idea is to move motion detection to the camera and keep object detection in frigate. Now, I either have to do both outside frigate or within frigate.

**Reply by NickM-27** [reply]
right, however frigate doesn't run object detection on the whole frame. It uses the areas where motion are detected to know where to run object detection.

**Reply by luisiam** [reply]
Is it possible to pass that information via the API call?

**Reply by NickM-27** [reply]
no, and that would likely be too slow compared to when frigate runs it locally

**Reply by luisiam** [reply]
I see. Thanks for the feedback.
